graphql_compiler/compiler/emit_sql.py

```diff
@@ -1131,6 +1179,11 @@ def fold(self, fold_scope_location: FoldScopeLocation) -> None:
         join_descriptor = self._sql_schema_info.join_descriptors[self._current_classname][
             full_edge_name
         ]
+        if not isinstance(join_descriptor, DirectJoinDescriptor):
+            raise NotImplementedError(
+                f"Edge {full_edge_name} is backed by a CompositeJoinDescriptor, "
+                "so it can't be used with @fold."
```

Comment:
Same reasoning as above, here's the equivalent error message here that follows my template from above:
```suggestion
                f"Edge {vertex_field} requires a JOIN across a composite key, which is currently "
                f"not supported for use with @fold."
```